xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/main/java/org/xwiki/refactoring/internal/job/DeleteJob.java

```diff
@@ -111,16 +126,24 @@ public void visit(DocumentReference documentReference)
 
     private void maybeDelete(DocumentReference documentReference)
     {
-        EntitySelection entitySelection = concernedEntities.get(documentReference);
+        Boolean skipRecycleBinProperty = this.getRequest().getProperty(SKIP_RECYCLE_BIN_PROPERTY);
+        boolean skipRecycleBin = this.configuration.canSkipRecycleBin()
+                                     && this.documentAccessBridge.isAdvancedUser()
+                                     && ObjectUtils.defaultIfNull(skipRecycleBinProperty, false);
+        EntitySelection entitySelection = this.concernedEntities.get(documentReference);
         if (entitySelection != null && !entitySelection.isSelected()) {
             // TODO: handle entitySelection == null which means something is wrong
             this.logger.info("Skipping [{}] because it has been unselected.", documentReference);
         } else if (!this.modelBridge.exists(documentReference)) {
             this.logger.warn("Skipping [{}] because it doesn't exist.", documentReference);
         } else if (!hasAccess(Right.DELETE, documentReference)) {
             this.logger.error("You are not allowed to delete [{}].", documentReference);
-        } else {
+        } else if (!skipRecycleBin) {
             this.modelBridge.delete(documentReference);
+            this.logger.debug("[{}] has been successfully moved to the recycle bin.", documentReference);
+        } else {
+            this.modelBridge.delete(documentReference, true);
+            this.logger.debug("[{}] has been successfully removed.", documentReference);
```

Comment:
"deleted" maybe to use the same terminology?

Reply:
Fixed https://github.com/xwiki/xwiki-platform/pull/1401/files#diff-57f9752a12af2ec241f5cb8a244f427fR146